### ISSUE TITLE
Incorrect cgroup when using systemd cgroup driver

### ISSUE THREAD

**Issue by derekwaynecarr** [kubernetes-incubator/cri-o#842]
Steps to reproduce:

```
# verify systemd cgroup driver
]$ cat /etc/crio/crio.conf | grep cgroup_manager
# cgroup_manager is the cgroup management implementation to be used
cgroup_manager = "systemd"
# spin up kubernetes with matching driver
$ cd kubernetes
$ export CGROUP_DRIVER=systemd
$ export KUBE_ENABLE_CLUSTER_DASHBOARD=true
$ export KUBE_ENABLE_CLUSTER_DNS=true
$ hack/local-up-cluster.sh
..
$ kubectl get pods --all-namespaces
# should see the dns and dashboard pods running
```

Verify the cgroup tree:

kubelet creates the following pod cgroup:

```
/sys/fs/cgroup/cpu/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod12723d69_9268_11e7_84d0_c85b76cda386.slice
```

note: kubepods-burstable-pod12723d69_9268_11e7_84d0_c85b76cda386.slice

there is nothing in the cgroupfs for the container, CRI-O is creating them here:

```
/sys/fs/cgroup/cpu/kubepods.slice.slice/kubepods.slice-kubepods_burstable.slice.slice/kubepods.slice-kubepods_burstable.slice-kubepods_burstable_pod12723d69_9268_11e7_84d0_c85b76cda386.slice.slice
```

note: the double encoding everywhere in the hierarchy

i think crio when running with cgroup manager = systemd should error when given invalid cgroups, but it should just convert from cgroupfs to systemd naming doing something similar to here:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cgroup_manager_linux.go#L109

**Comment by derekwaynecarr** [comment]
CRI-O should error if its configured for systemd and it gets a cgroup parent that does not end in .slice as well (didnt test if it does that now).

**Comment by runcom** [comment]
I never hit this issue, I'll work in this tomorrow, I may know where this is happening already.

**Comment by runcom** [comment]
Oh and we do already use that kubernetes function actually

**Comment by runcom** [comment]
I have a fix for this coming.

**Comment by runcom** [comment]
I didn't know one could set cgroup manager from kube. Right we assume cgroup parents are cgroupsfs so when you set systemd, it blows up when cri-o receives a systemd cgroup parent.

**Comment by runcom** [comment]
@derekwaynecarr could you verify that if you run kubelet with cgroupsfs and cri-o with systemd, cri-o does the conversion from cgroups to systemd correctly? (you said cri-o needs to convert when it's using systemd and receives a cgroupfs parent in your first comment)

**Comment by runcom** [comment]
@derekparker you're using an old master to test crio....please update it

**Comment by runcom** [comment]
**TL;DR;** CRI-O + #814 + #846 fixes all the cgroup issues :) (notice we need **both** PRs)

| Kubelet  | Docker | works? | cgroup |
| ------------- | ------------- | ------------- | ------------- |
| systemd  | systemd  | yes | `/sys/fs/cgroup/cpu/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod90d81341_92de_11e7_8cf2_507b9d4141fa.slice/docker-2227daf62df6694645fee5df53c1f91271546a9560e8600a525690ae252b7f63.scope/`
| cgroupfs  | cgroupfs  | yes, but no `docker-` prefix | `/sys/fs/cgroup/cpu/kubepods/burstable/pod941dc160-92df-11e7-b27e-507b9d4141fa/2b4d747eae5ed34775c069d913c81833b67a0148e731a36535c9f3b97ebfb87f/`
| systemd  | cgroupfs  | no | kubelet throws an error `Error: failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "systemd" is different from docker cgroup driver: "cgroupfs"`
| cgroupfs  | systemd  | no | kubelet throws an error `Error: failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "cgroupfs" is different from docker cgroup driver: "systemd"` 

| Kubelet  | CRI-O + #814 | works? | cgroup |
| ------------- | ------------- | ------------- | ------------- |
| systemd  | systemd  | yes but wrong cgroup | **kubelet cgroup**: `/sys/fs/cgroup/cpu/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod945ff58e_92e0_11e7_a0d6_507b9d4141fa.slice/` - **CRI-O cgroup (double `.slice.slice`)**: `/sys/fs/cgroup/cpu/kubepods.slice.slice/kubepods.slice-kubepods_burstable.slice.slice/kubepods.slice-kubepods_burstable.slice-kubepods_burstable_pod945ff58e_92e0_11e7_a0d6_507b9d4141fa.slice.slice/crio-4ca5c7010dffecea158ce82072db8b8ef211e38c3e5c1c7d4fe14024a7c89e4c.scope`
| cgroupfs  | cgroupfs  | yes | `/sys/fs/cgroup/cpu/kubepods/burstable/poda3cb7007-92e1-11e7-a18c-507b9d4141fa/crio-e2bce3aa1dac117b5ffa29318bdc273f55bb555dcbe684b31dd05f1e573bb3e5`
| systemd  | cgroupfs  | **WTF wrong cgroup** | `/sys/fs/cgroup/cpu/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod4e839739_92e2_11e7_93d7_507b9d4141fa.slice/crio-conmon-b1e75da956b6888f16e616254bfaf5d3eaee893087111d65507a6079dd388604/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod4e839739_92e2_11e7_93d7_507b9d4141fa.slice/crio-b1e75da956b6888f16e616254bfaf5d3eaee893087111d65507a6079dd388604`
| cgroupfs  | systemd  | yes **(???)** | `/sys/fs/cgroup/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podd553858f_92e2_11e7_bf60_507b9d4141fa.slice/crio-038a516ff55008ebc461ac7ec0d789d61c351b9d5fa753b2394f32af47b9634c.scope`

| Kubelet  | CRI-O + #846  | works? | cgroup |
| ------------- | ------------- | ------------- | ------------- |
| systemd  | systemd  | yes | `/sys/fs/cgroup/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podba678164_92e3_11e7_b480_507b9d4141fa.slice/crio-40db63c45b57ee9e57aa865674bb35510a86bb3560352bc478ad283ebc3811b8.scope`
| cgroupfs  | cgroupfs  | yes, but no `crio-` prefix, fix is in #814 | `/sys/fs/cgroup/kubepods/burstable/pod0dd4700f-92e4-11e7-be65-507b9d4141fa/364244572dc75829a1e55255cda23b5e4de82cee635475822d121d4a9da04892`
| systemd  | cgroupfs  | no | CRI-O throws an error `E0906 11:22:43.874859    7385 pod_workers.go:182] Error syncing pod b0386dc0-92e4-11e7-9540-507b9d4141fa ("kube-dns-7797cb8758-7xf9q_kube-system(b0386dc0-92e4-11e7-9540-507b9d4141fa)"), skipping: failed to "CreatePodSandbox" for "kube-dns-7797cb8758-7xf9q_kube-system(b0386dc0-92e4-11e7-9540-507b9d4141fa)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kube-dns-7797cb8758-7xf9q_kube-system(b0386dc0-92e4-11e7-9540-507b9d4141fa)\" failed: rpc error: code = Unknown desc = cri-o configured with cgroupfs cgroup manager, but received systemd slice as parent: kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podb0386dc0_92e4_11e7_9540_507b9d4141fa.slice/"`
| cgroupfs  | systemd  | no | CRI-O throws an error `E0906 11:25:46.040691   11395 pod_workers.go:182] Error syncing pod 4374ca63-92e5-11e7-b542-507b9d4141fa ("kube-dns-7797cb8758-r4l59_kube-system(4374ca63-92e5-11e7-b542-507b9d4141fa)"), skipping: failed to "CreatePodSandbox" for "kube-dns-7797cb8758-r4l59_kube-system(4374ca63-92e5-11e7-b542-507b9d4141fa)" with CreatePodSandboxError: "CreatePodSandbox for pod \"kube-dns-7797cb8758-r4l59_kube-system(4374ca63-92e5-11e7-b542-507b9d4141fa)\" failed: rpc error: code = Unknown desc = cri-o configured with systemd cgroup manager, but did not receive slice as parent: /kubepods/burstable/pod4374ca63-92e5-11e7-b542-507b9d4141fa"`
